libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1829,6 +1829,16 @@ void TypeChecker::endVisit(BinaryOperation const& _operation)
 				)
 			);
 	}
+	if (
+		TokenTraits::isCompareOp(_operation.getOperator()) &&
+		commonType->category() == Type::Category::Contract
+	)
+		m_errorReporter.warning(
+			9170_error,
+			_operation.location(),
+			"Comparison of variables of contract type will be deprecated in the next breaking version."
+			"Consider using an explicit cast to address type."
```

Comment:
If this is the same semantic behavior, you could also emit

```suggestion
			"Instead, use an explicit cast to address type."
```